data_steward/analytics/rt_cdr_qc/cdr_deid_qa_report7_cope_survey.py

```diff
@@ -106,71 +130,220 @@
+# +
 query = f'''
 
-SELECT COUNT (*) AS n_row_not_pass  
-FROM `{project_id}.{deid_cdr}.observation` 
-WHERE observation_concept_id IN  (37311060, 45763724) 
+SELECT observation_concept_id, concept_name,concept_code,vocabulary_id,observation_source_concept_id,COUNT(1) AS n_row_not_pass FROM 
+`{project_id}.{deid_cdr}.observation` ob
+JOIN `{project_id}.{deid_cdr}.concept` c
+ON ob.observation_concept_id=c.concept_id
+WHERE observation_concept_id IN  (37311060, 45763724) OR observation_source_concept_id IN  (37311060, 45763724)
+GROUP BY 1,2,3,4,5
+ORDER BY n_row_not_pass DESC
 
 '''
 df1=pd.read_gbq(query, dialect='standard')
-if df1.loc[0].sum()==0:
- df = df.append({'query' : 'Query5 COVID concepts suppression in observation table', 'result' : 'PASS'},  
+
+if df1['n_row_not_pass'].sum()==0:
+ df = df.append({'query' : 'Query5 COVID concepts suppression in observation table', 'result' : ''},  
                 ignore_index = True) 
 else:
- df = df.append({'query' : 'Query5 COVID concepts suppression in observation table', 'result' : ''},  
+ df = df.append({'query' : 'Query5 COVID concepts suppression in observation table' , 'result' : 'PASS'},  
+                ignore_index = True) 
+df1
+# -
+
+# # 6 Verify these concepts are NOT suppressed in EHR observation
+#   
+# [DC-1747]
+# these concepts 1333015, 	1333023	are not longer suppressed
+#
+# 1332737, [DC-1665] 
+#
+# 1333291
+#
+# 1332904,1333140 should be generalized to 1332737
+#
+# 1332843 should be generalized. 
+
+# +
+query = f'''
+
+SELECT observation_source_concept_id, concept_name,concept_code,vocabulary_id,observation_concept_id,COUNT(1) AS n_row_not_pass FROM 
+`{project_id}.{deid_cdr}.observation` ob
+JOIN `{project_id}.{deid_cdr}.concept` c
+ON ob.observation_source_concept_id=c.concept_id
+WHERE observation_source_concept_id IN  (1333015, 1333023, 1332737,1333291,1332904,1333140,1332843) OR observation_concept_id IN  (1333015, 1333023,1332737,1333291,1332904,1333140,1332843 )
+GROUP BY 1,2,3,4,5
+ORDER BY n_row_not_pass DESC
+
+'''
+df1=pd.read_gbq(query, dialect='standard')
+
+if df1['n_row_not_pass'].sum()==0:
+ df = df.append({'query' : 'Query6 The concepts are not suppressed in observation table', 'result' : ''},  
+                ignore_index = True) 
+else:
+ df = df.append({'query' : 'Query6 The concepts are not suppressed in observation table' , 'result' : 'PASS'},  
+                ignore_index = True) 
+df1
+# -
+
+# # 7 Vaccine-related concepts as these EHR-submitted COVID concepts are disallowed from RT 
+#
+# DC-1752
+#
+
+# +
+
+query = f'''
+DECLARE vocabulary_tables DEFAULT ['vocabulary', 'concept', 'source_to_concept_map', 
+                                   'concept_class', 'concept_synonym', 'concept_ancestor',
+                                   'concept_relationship', 'relationship', 'drug_strength'];
+
+DECLARE query STRING;
+
+CREATE OR REPLACE TABLE `aou-res-curation-prod.R2021q3r1_deid_sandbox.concept_usage` (
```

Comment:
Hi, @jqian2015 Dataset names were hardcoded in this query, it needs to be parameterized.

Reply:
i changed and updated. thanks.